From aae809e09868ff0c539a12ecfe25f8fb5b86e5d7 Mon Sep 17 00:00:00 2001
From: Hu Bingquan <bingquan.hu@intel.com>
Date: Thu, 3 Mar 2016 22:37:08 +0800
Subject: [PATCH] power HAL: send uevent when register power_HAL_suspend

if related driver is built as module, no uevent is sent to
ueventd to change power_HAL_suspend's permission.

analysis:
first uevnet is sent when device_add(), uevnetd can receive
this uevent. but power_HAL_suspend is not exist.
when driver is loaded, power_HAL_suspend is created. but no
uevent to notify uevnetd to change the premission.

fix:
send KOBJ_CHANGE uevent after power_HAL_suspend is created.

Same issue on Sofia 3GR. See
https://jira01.devtools.intel.com/browse/GMINL-13671

Change-Id: Ibfbf5d0c81a2a784a5419de18b71aa3ff3acad8c
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-25984
Signed-off-by: Zhang Ning <ning.a.zhang@intel.com>
Signed-off-by: Li Ting <ting.li@intel.com>
Signed-off-by: Hu Bingquan <bingquan.hu@intel.com>
Reviewed-on: https://android.intel.com:443/433070
Reviewed-on: https://android.intel.com:443/458083
Reviewed-on: https://android.intel.com:443/477324
---
 kernel/power/main.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/kernel/power/main.c b/kernel/power/main.c
index bedba2e..25953c3 100644
--- a/kernel/power/main.c
+++ b/kernel/power/main.c
@@ -233,7 +233,6 @@ static int __init pm_debugfs_init(void)
 
 late_initcall(pm_debugfs_init);
 #endif /* CONFIG_DEBUG_FS */
-
 #endif /* CONFIG_PM_SLEEP */
 
 #ifdef CONFIG_PM_SLEEP_DEBUG
@@ -635,11 +634,16 @@ static inline int pm_start_workqueue(void) { return 0; }
 
 int register_power_hal_suspend_device(struct device *dev)
 {
+	int ret;
 	if (!power_hal_kobj || !dev)
 		return -ENODEV;
 
-	return sysfs_create_link(power_hal_kobj, &dev->kobj,
+	ret = sysfs_create_link(power_hal_kobj, &dev->kobj,
 			dev_name(dev));
+
+	kobject_uevent(&dev->kobj, KOBJ_CHANGE);
+
+	return ret;
 }
 EXPORT_SYMBOL(register_power_hal_suspend_device);
 
-- 
1.9.1

